packages/charts/src/chart_types/xy_chart/state/selectors/get_api_scale_configs.ts

```diff
@@ -54,19 +53,17 @@ export function getScaleConfigsFromSpecs(
   seriesSpecs: BasicSeriesSpec[],
   settingsSpec: SettingsSpec,
 ): ScaleConfigs {
-  const isHorizontalChart = isHorizontalRotation(settingsSpec.rotation);
-
   // x axis
-  const xAxes = axisSpecs.filter((d) => isHorizontalChart === isHorizontalAxis(d.position));
-  const xTicks = xAxes.reduce<number>((acc, { ticks = X_SCALE_DEFAULT.desiredTickCount }) => {
+  const xAxesSpecs = axisSpecs.filter((spec) => isXDomain(spec.position, settingsSpec.rotation));
+  const maxTickCountForXAxes = xAxesSpecs.reduce<number>((acc, { ticks = X_SCALE_DEFAULT.desiredTickCount }) => {
     return Math.max(acc, ticks);
-  }, X_SCALE_DEFAULT.desiredTickCount);
+  }, -Infinity);
```

Comment:
```suggestion
  }, Number.MIN_SAFE_INTEGER);
```

Didn't @monfera say something about using `Number.MIN_SAFE_INTEGER` over `Infinity`?

Reply:
I think he was referring to the usage of non finite number when sorting:
```
const BIG_NUMBER = Number.MAX_SAFE_INTEGER; // the sort comparator must yield finite results, can't use infinities

```

Reply:
I think we could also simply put a `-1` there and check for a positive value when assigning the value to the ScaleConfig

Reply:
Cool, either way is fine.